server/service/src/repack/query.rs

```diff
@@ -47,36 +47,87 @@ pub fn get_repack(ctx: &ServiceContext, invoice_id: &str) -> Result<Repack, Repo
     })
 }
 
+pub fn get_repacks_by_stock_line(
+    ctx: &ServiceContext,
+    stock_line_id: &str,
+) -> Result<Vec<Repack>, RepositoryError> {
+    let connection = &ctx.connection;
+
+    let invoices = InvoiceRepository::new(connection).query_by_filter(
+        InvoiceFilter::new()
+            .store_id(EqualFilter::equal_to(&ctx.store_id))
+            .r#type(InvoiceRowType::Repack.equal_to())
+            .stock_line_id(stock_line_id.to_owned()),
+    )?;
+
+    let mut repacks = Vec::new();
+
+    for invoice in invoices {
+        let invoice_lines = InvoiceLineRepository::new(connection).query_by_filter(
+            InvoiceLineFilter::new().invoice_id(EqualFilter::equal_to(&invoice.invoice_row.id)),
```

Comment:
Hmm, although it's not likely that a single stock line would have lots of repacks, so it's ok to have multiple queries for invoice lines, usually I would try to reduce database operations (get all invoice lines for invoices and then find the line for invoice)